### PR TITLE
[Migrations] Support PostgreSQL migrations

### LINKED ISSUE #11851: Unable to do a fresh install with postgres
**Sylius version affected**: 1.8.0

**Description**  

See https://github.com/Sylius/Sylius/issues/8853

**Steps to reproduce**  

Follow the install step with a PostgreSQL server instead of MySQL.

**Possible Solution**  

Go back to the old migration system where Migration files were in src/Migrations instead of each package.

### PR DESCRIPTION
**Pull request by Rafikooo**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13                   |
| Bug fix?        | yes                                                       |
| New feature?    | no                                                      |
| BC breaks?      | no                                                      |
| Deprecations?   | no |
| Related tickets | fixes https://github.com/Sylius/Sylius/issues/11851  #mentioned https://github.com/Sylius/Sylius/pull/14925          |
| License         | MIT                                                          |

This PR introduced the initial migrations dedicated to `PostgreSQL` database. From now on, each new migration should be created for both databases and extend the corresponding `AbstractMigration` class

### CODE REVIEW

File: .github/workflows/ci_e2e-pgsql.yaml [hunk 1]
```diff
@@ -53,7 +53,7 @@ jobs:
                     restore-keys: "${{ github.run_id }}-${{ runner.os }}-${{ hashFiles('composer.json') }}-php-${{ matrix.php }}-symfony-${{ matrix.symfony }}-"
 
             -   name: Build application
-                uses: jakubtobiasz/SyliusBuildTestAppAction@v2.0
+                uses: Rafikooo/SyliusBuildTestAppAction@postgres
```
Comment (Rafikooo) [inline .github/workflows/ci_e2e-pgsql.yaml]: https://github.com/Rafikooo/SyliusBuildTestAppAction/commit/c9df907a9fe717162e11b81675e2fd23b85544c6

Comment (jakubtobiasz) [reply .github/workflows/ci_e2e-pgsql.yaml]: I'll update `SyliusBuildTestAppAction` on `SyliusLabs` today or tomorrow. So don't merge until I do it.

**Review by jakubtobiasz — APPROVED** [review]

File: src/Sylius/Bundle/CoreBundle/EventListener/PostgreSQLDefaultSchemaListener.php
Comment (Rafikooo) [inline src/Sylius/Bundle/CoreBundle/EventListener/PostgreSQLDefaultSchemaListener.php:1]: Without this listener, any `doctrine:migrations:diff` creates such a migration
image

File: src/Sylius/Bundle/CoreBundle/Migrations/Version20230420151332.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        $this->addSql('ALTER TABLE sylius_promotion_translation ADD CONSTRAINT FK_3C7A76182C2AC5D3 FOREIGN KEY (translatable_id) REFERENCES sylius_promotion (id) ON DELETE CASCADE NOT DEFERRABLE INITIALLY IMMEDIATE');
+        $this->addSql('ALTER TABLE sylius_channel ADD channel_price_history_config_id INT DEFAULT NULL');
+        $this->addSql('ALTER TABLE sylius_channel ADD CONSTRAINT FK_16C8119E75F20EAE FOREIGN KEY (channel_price_history_config_id) REFERENCES sylius_channel_price_history_config (id) ON DELETE CASCADE NOT DEFERRABLE INITIALLY IMMEDIATE');
+        $this->addSql('CREATE UNIQUE INDEX UNIQ_16C8119E75F20EAE ON sylius_channel (channel_price_history_config_id)');
+        $this->addSql('ALTER TABLE sylius_channel_pricing ADD lowest_price_before_discount INT DEFAULT NULL');
+        $this->addSql('ALTER TABLE sylius_user_oauth ALTER access_token TYPE TEXT');
+        $this->addSql('ALTER TABLE sylius_user_oauth ALTER access_token TYPE TEXT');
+        $this->addSql('ALTER TABLE sylius_user_oauth ALTER refresh_token TYPE TEXT');
+        $this->addSql('ALTER TABLE sylius_user_oauth ALTER refresh_token TYPE TEXT');
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/CoreBundle/Migrations/Version20230420151332.php]:
```suggestion
        $this->addSql('ALTER TABLE sylius_user_oauth ALTER access_token TYPE TEXT');
        $this->addSql('ALTER TABLE sylius_user_oauth ALTER refresh_token TYPE TEXT');
```

**Review by NoResponseMate — APPROVED** [review]

**Comment by NoResponseMate** [comment]
Thank you, @Rafikooo!

**Pull request merged**